packages/rum-react/test/specs/get-apm-route.spec.js

```diff
@@ -44,8 +44,16 @@ function Component(props) {
 }
 
 describe('ApmRoute', function() {
+  let serviceFactory, configService
+
+  beforeEach(() => {
+    serviceFactory = createServiceFactory()
+    configService = serviceFactory.getService('ConfigService')
+    configService.init({ active: true })
```

Comment:
Is this required? I mean `active: true` since just calling the init should be enough, active is true by default.
If the tests fail without this, then we have introduced a bug!

Reply:
I removed the `apm.init` in favor of this and yes we require this to be set since now we expect the isActive function to return true to start  instrumenting the route https://github.com/elastic/apm-agent-rum-js/blob/9f429dbf6d39e7208ca926140eeec839c5be7a8b/packages/rum-react/src/get-with-transaction.js#L51-L53

Reply:
What is the advantage of this approach vs just using `apm.init`? IMO, apm.init makes more sense since we're passing the apm instance into `get-with-transaction`

Reply:
Two things
1. We need to call init on all the tests since we expect the config.isActive to be true to make it work. 
2. `apm.init` does lots of things under the hood like registering error listeners, config validation, perf monitoring which is not required for these specific tests.

Reply:
For No. 1 we still need to call init on ConfigService, I don't see a difference here. 

I think since ApmBase is a dependency to `get-with-transaction` therefore it needs to be initialised for the test to be actually realistic. e.g. the things that ApmBase does in init might actually affect these tests!

Reply:
Are you saying that we have to call apm.init for all of the tests instead of config.init? 

We can do that but we have to stop error listened and onload listeners from initializing otherwise we would have side effects

Reply:
We don't have to call apm.init for all the tests but I'm ok with calling the init for all of them as well if that makes the tests easier to write.

For the side effects we probably can use flags to disable page load and error handling!

Reply:
Yeah that would be equal to what I was going for, Let's do that then.

Reply:
I have changed it now, Please have a look once @jahtalab